crawl-ref/source/traps.cc

```diff
@@ -457,13 +457,18 @@ vector<coord_def> find_golubria_on_level()
     return ret;
 }
 
-static bool _find_other_passage_side(coord_def& to)
+static bool _find_other_passage_side(coord_def& to, bool& is_blocked)
```

Comment:
I don't love adding an out-parameter here. The two booleans are particularly confusing because you set `is_blocked` if *any* of the passages were blocked, which is something the caller doesn't care about at all: they only want to know if all passages were blocked, and therefore the trap couldn't trigger.

Returning a new enum type instead like PASSAGE_FOUND`/`PASSAGE_BLOCKED`/`PASSAGE_NONE` or something (feel free to choose better names) would be clearer than having two booleans.